kubernetes/autoscaler
# Support attribute-based instance selection for AWS

#### Which component this PR applies to?


#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Support attribute-based instance selection using instance requirements.

This PR updates aws-sdk-go from v1.38.49 to v1.42.25 in order to pick up instance requirements feature changes.

#### Which issue(s) this PR fixes:

Fixes #4479 

#### Special notes for your reviewer:

Testing done using an ASG w/ pre-existing instances:
1. Created an ASG w/ Spot and CapacityOptimized allocation strategy
2. Set InstanceRequirements Overrides w/ vcpu min: 10 max: 100
3. Update Cluster Autoscaler Min size to 1, Max 10
4. Waited for an instance to be added with Min 10 vcpu
5. Update InstanceRequirements Overrides to vcpu min: 2, max: 100
6. Cordon the 1st Node that came up in the ASG so that we can differentiate between nodes from step 2 vs step 5
7. Create a deployment with pods with resource request of 1 vcpu each
8. Scale deployment to 5 replicas
9. Expectation and observed: CA raises Desired Capacity to 6 under newer requirements, vs 1-2 expected nodes if under step 2 requirements 

#### Does this PR introduce a user-facing change?


```release-note
Added support for attribute-based instance type selection for AWS using available instance requirements.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:


```docs

```
